### PR TITLE
GRF - Async Facilitator Lambda - Data Access Layer 

### PR DESCRIPTION
**Pull request by kamyarz-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

https://app.asana.com/0/1202938635037719/1204131945299326/f

### CODE REVIEW

File: source/refreezer/application/db_accessor/dynamoDb_accessor.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    def get_item(self, key: Dict[str, Any]) -> Optional[Dict[str, Any]]:
+        response = self.table.get_item(Key=key)
+        return response.get("Item")
+
+    def update_item(
+        self,
+        key: Dict[str, Any],
+        update_expression: str,
+        expression_attribute_values: Dict[str, Any],
+    ) -> None:
```
Comment (simonkrol) [inline source/refreezer/application/db_accessor/dynamoDb_accessor.py:42]: Should we be looking for failures in these sorts of calls(insert/update/delete)? Ie: return True/False from this method, depending on whether the update actually went through?

File: source/refreezer/application/db_accessor/dynamoDb_accessor.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+class DynamoDBAccessor:
+    # TODO: Will delete region value later. The fix is in place in another branch
+    def __init__(self, table_name: str, region: str = "us-east-1") -> None:
+        self.dynamodb: DynamoDBServiceResource = boto3.resource("dynamodb", region)
+        self.table: Table = self.dynamodb.Table(table_name)
+
+    def insert_item(self, item: Dict[str, Any]) -> None:
+        self.table.put_item(Item=item)
+        logger.info("Succesfully inserted item into the database")
```
Comment (bassemwanis) [inline source/refreezer/application/db_accessor/dynamoDb_accessor.py:31]: Typo "Successfully" in all logger.info messages

File: source/refreezer/application/db_accessor/dynamoDb_accessor.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    def delete_item(self, key: Dict[str, Any]) -> None:
+        self.table.delete_item(Key=key)
+        logger.info("Succesfully deleted item from the database")
+
+    def query_items(
+        self, primary_key: str, key_condition_expression: str
+    ) -> Optional[List[Dict[str, Any]]]:
+        response = self.table.query(
+            KeyConditionExpression=Key(primary_key).eq(key_condition_expression),
```
Comment (bassemwanis) [inline source/refreezer/application/db_accessor/dynamoDb_accessor.py:58]: In the future, is it possible to expand this to handle other comparison operator?

File: source/refreezer/application/db_accessor/dynamoDb_accessor.py (hunk 1 above)
Comment (kamyarz-aws) [reply source/refreezer/application/db_accessor/dynamoDb_accessor.py:42]: Usually db function calls have exception handling. So I dont know if I should make sure of it here in the data access layer

File: source/refreezer/application/db_accessor/dynamoDb_accessor.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    from mypy_boto3_dynamodb.service_resource import Table
+else:
+    DynamoDBServiceResource = object
+    Table = object
+
+logger = logging.getLogger()
+
+
+class DynamoDBAccessor:
+    # TODO: Will delete region value later. The fix is in place in another branch
```
Comment (bassemwanis) [inline source/refreezer/application/db_accessor/dynamoDb_accessor.py:24]: From this PR https://github.com/aws-solutions/amazon-s3-glacier-refreezer/pull/85, I believe you still need to pass the region to boto3.resource. Can you please confirm this with Simon and adjust this comment accordingly?

File: source/tests/unit/application/db_accessor/test_db_accessor.py [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    DynamoDBClient = object
+
+
+@pytest.fixture(autouse=True)
+def mock_env_variables() -> None:
+    os.environ["AsyncFacilitatorTableName"] = "RandomTable"
+    os.environ["AWS_REGION"] = "us-east-1"
+
+
+@pytest.fixture()
```
Comment (simonkrol) [inline source/tests/unit/application/db_accessor/test_db_accessor.py:33]: Like we chatted about in our meeting, you can add a new fixture to the tests/unit/conftest.py for the dynamodb_resource and use it as part of this fixture, that will let you get rid of line 35 here.

**Review by bassemwanis — APPROVED** [review]

**Review by simonkrol — APPROVED** [review]

**Pull request merged**